site/src/pages/TemplateSettingsPage/TemplatePermissionsPage/TemplatePermissionsPage.tsx

```diff
@@ -1,29 +1,36 @@
 import Button from "@mui/material/Button";
 import Link from "@mui/material/Link";
 import ArrowRightAltOutlined from "@mui/icons-material/ArrowRightAltOutlined";
-import { useMachine } from "@xstate/react";
 import { Paywall } from "components/Paywall/Paywall";
 import { Stack } from "components/Stack/Stack";
 import { useFeatureVisibility } from "hooks/useFeatureVisibility";
 import { useOrganizationId } from "hooks/useOrganizationId";
 import { FC } from "react";
 import { Helmet } from "react-helmet-async";
 import { pageTitle } from "utils/page";
-import { templateACLMachine } from "xServices/template/templateACLXService";
 import { useTemplateSettings } from "../TemplateSettingsLayout";
 import { TemplatePermissionsPageView } from "./TemplatePermissionsPageView";
 import { docs } from "utils/docs";
+import { useMutation, useQuery, useQueryClient } from "react-query";
+import { setGroupRole, setUserRole, templateACL } from "api/queries/templates";
+import { displaySuccess } from "components/GlobalSnackbar/utils";
 
 export const TemplatePermissionsPage: FC<
   React.PropsWithChildren<unknown>
 > = () => {
   const organizationId = useOrganizationId();
   const { template, permissions } = useTemplateSettings();
   const { template_rbac: isTemplateRBACEnabled } = useFeatureVisibility();
-  const [state, send] = useMachine(templateACLMachine, {
-    context: { templateId: template.id },
-  });
-  const { templateACL, userToBeUpdated, groupToBeUpdated } = state.context;
+  const templateACLQuery = useQuery(templateACL(template.id));
+  const queryClient = useQueryClient();
+
+  const addUserMutation = useMutation(setUserRole(queryClient));
+  const updateUserMutation = useMutation(setUserRole(queryClient));
+  const removeUserMutation = useMutation(setUserRole(queryClient));
+
+  const addGroupMutation = useMutation(setGroupRole(queryClient));
+  const updateGroupMutation = useMutation(setGroupRole(queryClient));
+  const removeGroupMutation = useMutation(setGroupRole(queryClient));
```

Comment:
Do we need the three copies so we can have separate errors for each?  Although, it does not look like we are showing the errors anywhere?  I am not sure we were showing errors before though, so maybe we will not need to handle them in this PR.

Reply:
> Do we need the three copies so we can have separate loaders/errors for each? 

Exactly.

> Although, it does not look like we are showing the errors anywhere? I am not sure we were showing errors before though, so maybe we will not need to handle them in this PR.

Yes, we didn't handle errors before but it is something on my radar